src/analysis/classic/warlock/demonology/modules/spells/Doomguard.tsx

```diff
@@ -0,0 +1,394 @@
+              relatedBuffs: [tailoring.LIGHTWEAVE_BUFF_RANK_2],
+              issueWithItemOrEnchant: !lighweaveAvailable,
+              snapshotSummary: lighweaveAvailable ? (
+                <></>
+              ) : (
+                <li>
+                  <ItemLink id={equippedItem.id} details={equippedItem} icon={false}>
+                    <Icon icon={equippedItem.icon} />
+                  </ItemLink>{' '}
+                  is not enchanted with <SpellLink spell={tailoring.LIGHTWEAVE_BUFF_RANK_2} />. If
+                  you are not tailor, consider to change profession. This is the 2nd best
+                  profession, especially as a warlock due to snapshot mechanics.
+                </li>
+              ),
+            });
+          }
+          break;
+        case GEAR_SLOTS.MAINHAND:
+          {
+            const powerTorrentAvailable =
+              equippedItem.permanentEnchant === enchanting.POWER_TORRENT_BUFF.effectId;
+            this.snapshotQualityEntries.push({
+              item: equippedItem,
+              snapshotQuality: QualitativePerformance.Fail,
+              relatedBuffs: [enchanting.POWER_TORRENT_BUFF],
+              issueWithItemOrEnchant: !powerTorrentAvailable,
+              snapshotSummary: powerTorrentAvailable ? (
+                <></>
+              ) : (
+                <li>
+                  <ItemLink id={equippedItem.id} details={equippedItem} icon={false}>
+                    <Icon icon={equippedItem.icon} />
+                  </ItemLink>{' '}
+                  is not enchanted with <SpellLink spell={enchanting.POWER_TORRENT_BUFF} />. This is
+                  by far the best weapon enchant, especially as a warlock due to snapshot mechanics.
+                </li>
+              ),
+            });
+          }
+          break;
+      }
+    });
+
+    // Volcanic potion
+    this.snapshotQualityEntries.push({
+      snapshotQuality: QualitativePerformance.Fail,
+      issueWithItemOrEnchant: false,
+      relatedBuffs: [potions.VOLCANIC_POTION],
+      snapshotSummary: <></>,
+    });
+
+    // Prepull hurricane tracking
+    this.snapshotQualityEntries.push({
+      snapshotQuality: QualitativePerformance.Fail,
+      issueWithItemOrEnchant: false,
+      relatedBuffs: [enchanting.HURRICANE_BUFF],
+      snapshotSummary: <></>,
+    });
+
+    // TODO Check if  DS really interfer with DG ==> Meta clearly doesnt
+
+    // 12024 average normal hit with flask and food, no DS no meta
+    // 12024 average normal hit with flask and food, no DS but meta
+    // 14467 average normal hit with flask and food, DS no meta
+  }
+  private onDoomguardSummonEvent(event: CastEvent) {
+    this.doomguardSummonData.doomguard.summonEvent = event;
+    this.doomguardSummonData.doomguard.summonTimestamp = event.timestamp;
+
+    const doomguardId = this.owner.playerPets.find(
+      (pet) => pet.guid === 11859 && pet.petOwner === this.selectedCombatant.player.id,
+    )?.id;
```

Comment:
add a `const` rather than referencing a raw game id

```suggestion
    const doomguardId = this.owner.playerPets.find(
      (pet) => pet.guid === doomguardGameId && pet.petOwner === this.selectedCombatant.player.id,
    )?.id;
```

and before the `class`:

```tsx
const doomguardGameId = 11859;
```